src/core/batch/batch.ts

```diff
@@ -17,17 +20,49 @@ interface BloomFilterData {
   RowGroupIndex: number;
 }
 
+interface WriteBatchFileOptions {
+  bloomFilters?: BloomFilterOptions;
+}
+
 export type BatchFileObject = string;
 
 /**
- * createFile() takes a series of DSNP messages and returns a Batch file
- * object for publishing. This method is not yet implemented.
+ * createFile() takes a series of DSNP messages and returns a URL
+ * for storage location.
  *
- * @param events  An array of DSNPEvents to include in the batch file
- * @returns         A batch file object
+ * @param messages An array of DSNPMessage to include in the batch file
+ * @returns         A URL of the storage location
+ * @throws error if messages argument is empty.
  */
-export const createFile = (_events: DSNPType[]): BatchFileObject => {
-  throw NotImplementedError;
+export const createFile = async (targetPath: string, messages: DSNPMessage[]): Promise<URL> => {
+  if (messages.length === 0) throw EmptyArrayError;
+
+  const schema = new ParquetSchema(getSchemaFor(messages[0].dsnpType));
+  const bloomFilterOptions = getBloomFilterOptionsFor(messages[0].dsnpType);
+
+  return putStream(targetPath, async (writeStream: ReadWriteStream) => {
+    await writeBatch(writeStream, schema, messages, { bloomFilters: bloomFilterOptions });
+  });
+};
+
+/**
+ * writeBatch() takes a series of DSNP messages and returns a Batch file
+ * object for publishing.
+ *
+ * @param messages An array of DSNPMessage to include in the batch file
```

Comment:
Missing some `@param`s here as well. I wonder if we can add a linter for this. I have the same trouble.